### ISSUE TITLE
Hidden environments in getModuleList not working

### ISSUE THREAD

**Issue by timcdlucas** [zoonproject/zoon#389]
```
f <- function(){
  .zoonHidden <- new.env(parent = globalenv())
}

f()
exists('.zoonHidden', mode = 'environment')

[1] FALSE

```
This looks to be the solution but I haven't sorted it.
https://stackoverflow.com/questions/6375790/r-creating-an-environment-in-the-globalenv-from-inside-a-function

**Comment by goldingn** [comment]
Yeah, that's not assigning .zoonHidden to the global environment, you'd have to do that explicitly with assign() (CRAN dislike this, though not sure their checks will pick it up). 

Does it need to be in global though? Could it be in the package namespace instead? I.e. you could just do:
```r
.zoonHidden <- new.env()
```
in the package code, then it should be in the search path for every zoon function, but not visible to the user (unless they explicitly go digging in the namespace)

**Comment by timcdlucas** [comment]
Yes I think in package namespace is fine so this should work fine.

**Comment by Doi90** [comment]
Unless you've already fixed it, this should do the trick right? <<- instead of <- inside a function assigns to the global environment.

```f <- function(){
   .zoonHidden <<- new.env(parent = globalenv())
 }
f()
exists('.zoonHidden', mode = 'environment')
[1] TRUE```

**Comment by timcdlucas** [comment]
I'll try it, but `parent = globalenv()` bit is explicitely stating it should be created in the global environment anyway. 

I think saving stuff in the package namespace is probably the correct solution anyway.

**Comment by goldingn** [comment]
I've fixed this in [`goldingn/zoon@master`](https://github.com/goldingn/zoon/commit/716b6c9fd710e007c34c9c76345f09e8db5cc51f)

`parent = globalenv()` *doesn't* create the environment in the global environment it creates the new environment in this environment, but sets its parent environment as the global environment. The parent environment is the next environment in the search path, it needn't be the environment in which the environment is defined.

In any case, I just created `.zoonHidden` in the package namespace and deleted that clause.
